core/src/test/scala/com/pingcap/tispark/datasource/AddingIndexSuite.scala

```diff
@@ -0,0 +1,84 @@
+package com.pingcap.tispark.datasource
+
+import com.pingcap.tikv.exception.TiBatchWriteException
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+
+class AddingIndexSuite extends BaseDataSourceTest("adding_index") {
+  private val row1 = Row(1, 1, "Hello")
+  private val row2 = Row(2, 2, "TiDB")
+  private val row3 = Row(3, 3, "Spark")
+  private val row4 = Row(4, 4, "abde")
+  private val row5 = Row(5, 5, "Duplicate")
+
+  private val row2_v2 = Row(2, 2, "TiSpark")
+  private val row3_v2 = Row(3, 3, "TiSpark")
+  private val row4_v2 = Row(4, 4, "TiSpark")
+  private val row5_v2 = Row(5, 5, "TiSpark")
+
+  private val schema = StructType(
+    List(
+      StructField("pk", IntegerType),
+      StructField("i", IntegerType),
+      StructField("s", StringType)
+    )
+  )
+
+  test("pk is not handle adding unique index") {
+    dropTable()
+    jdbcUpdate(
+      s"create table $dbtable(pk int, i int, s varchar(128), unique index(i), primary key(s))"
+    )
+    jdbcUpdate(
+      s"insert into $dbtable values(1, 1, 'Hello')"
+    )
+    // insert row2 row3
+    tidbWrite(List(row2, row3), schema)
+    testTiDBSelect(Seq(row1, row2, row3))
+
+    intercept[TiBatchWriteException] {
+      // insert row2 row4
+      tidbWrite(List(row2, row4), schema)
```

Comment:
better to compare error message

Reply:
The error message may change frequently, I suggest not to do it right now.

Reply:
then please add a `TODO`